Wikipedia/Code/DiffRevisionAnimator.swift

```diff
@@ -0,0 +1,66 @@
+
+import Foundation
+
+protocol DiffRevisionAnimating: class {
+    var embeddedViewController: UIViewController? { get }
+    var animateDirection: DiffRevisionAnimator.Direction?  { get set }
+}
+
+class DiffRevisionAnimator : NSObject, UIViewControllerAnimatedTransitioning {
+    
+    enum Direction {
+        case up
+        case down
+    }
+    
+    let direction: Direction
+    
+    init(direction: Direction) {
+        self.direction = direction
+    }
+    
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+
+        let container = transitionContext.containerView
+
+        guard let fromView = transitionContext.view(forKey: UITransitionContextViewKey.from) else { return }
+        guard let toView = transitionContext.view(forKey: UITransitionContextViewKey.to) else { return }
+        guard let toVC = (transitionContext.viewController(forKey: UITransitionContextViewControllerKey.to) as? DiffRevisionAnimating) else { return }
+        
```

Comment:
Should these early returns complete the transition?

Reply:
agreed, fixed